explainer.md

```diff
@@ -58,28 +58,29 @@ class MyElement extends HTMLElement {
 	* Calling `replaceSync(text)` on a constructed stylesheet replaces the content of the stylesheet with `text` synchronously, but it doesn't allow any `@import` rules
 	* We can't insert `@import` rules with `insertRule(rule)` to constructed stylesheets.
 	* Example:
-	```js  
+	```js
 	// Fine, returns Promise that resolves when 'some.css' finished loading.
 	sheet.replace("@import('some.css');"); 
 	// Fails
 	sheet.replace("@import('some.css');"); 
 	sheet.insertRule("@import('some.css');"); 
 	```
 
-* Each constructed `CSSStyleSheet` is "tied" to the `Document` it is constructed on, meaning that it can only be used in that document tree (whether in a top-level document or shadow trees).
+
+* Each constructed `CSSStyleSheet` is "tied" to the `Document` it is constructed on, meaning that it can only be used in that document tree (whether in a top-level document or shadow trees). It can be adopted into a different document tree, but it will be ignored for style calculation purposes.
```

Comment:
So this means that the proposed behavior is to keep the `StyleSheet` objects in the `ShadowRoot`, but ignoring them during style recalc? That sounds like a huge pain to me.

Reply:
Hmm, interesting, can you explain more about why it's hard in Gecko? For blink the change is quite simple (see this [CL](https://chromium-review.googlesource.com/c/chromium/src/+/1375296)). Also, since this PR is merged already, it might be better to continue this discussion in a new issue on this repo. I can make one if you want, or you can start one with your perspective on this.

Reply:
(oops, linked the wrong CL, I edited it but in case you're reading from email, the CL is https://chromium-review.googlesource.com/c/chromium/src/+/1377486)

Reply:
I don't think it's hard to implement or such, but feels really weird that you can poke at the `cssRules` of an object and change it without having any effect whatsoever on the page.

Reply:
The spec already says we should throw if any of the sheets in the array are not constructed. Why not also throw if any of the sheets have a different constructor document?

Reply:
@lilles So the difference is for non-constructed stylesheets the entry point is only when adding to the `adoptedStyleSheets`, while for different-document one there's also the case of adopting a subtree that contains `adoptedStyleSheets` with stylesheets from another document tree. That means we need to change the "adopt a node" spec to to go through every shadow root and check each entry in their `adoptedStyleSheets`, which might be annoying.

Reply:
BTW I reopened #23, might want to continue there.

Reply:
Oh. Didn't think about that.